### ISSUE TITLE
Failed to open stream

### ISSUE THREAD

**Issue by rhandford** [mailgun/mailgun-php#723]
I have updated my mailgun-php from 3.0 to 3.1 but now I get the error

```
Uncaught Buzz\Exception\NetworkException:  
file_get_contents(https://api.mailgun.net/v3/sandbox68cae6f5cd0443838a8851b2ceff97cb.mailgun.org/messages):  
failed to open stream: operation failed in G:\Server\xampp\htdocs\dev\mgapi.new\vendor\kriswallsmith\buzz\lib\Client\FileGetContents.php:26
```

So I followed the instructions in 3.1build a new autoload and made the simple  file shown below.

```php
<?php declare(strict_types=1);  // ensure she-bang is included for strict type to be accepted.

require 'vendor/autoload.php';
use Mailgun\Mailgun;

/*
 * Sand box site constants sandbox only works on US url api endpoint
 */
define('MAILGUN_API_ENDPOINT','https://api.mailgun.net/v3');
define('API_CREDENTIALS_STRING','12345678901234567890123456789012345678912345670');
define('DEFAULT_DOMAIN','sandbox12345678901234567890123456789012345678912345670b.mailgun.org');

//Here's how to send a message using the SDK:

// First, instantiate the SDK with your API credentials
$mg = Mailgun::create(API_CREDENTIALS_STRING); // For US servers

$mg->messages()->send(DEFAULT_DOMAIN,[
  'from'    => 'spam@ldwa.org.uk',
  'to'      => 'emailsupport@ldwa.org.uk',
  'subject' => 'The PHP SDK is awesome!',
  'text'    => 'It is so simple to send a message.'
]);
```

But I am still getting the same error below.  Can some tell me what has change and what do I need to change to resolve this.

```
PHP Fatal error:  Uncaught Buzz\Exception\NetworkException: file_get_contents(https://api.mailgun.net/v3/sandbox12345678901234567890123456789012345678912345670b.mailgun.org/messages): failed to open stream: operation failed in G:\Server\xampp\htdocs\dev\mgapi.new\vendor\kriswallsmith\buzz\lib\Client\FileGetContents.php:26
Stack trace:
#0 G:\Server\xampp\htdocs\dev\mgapi.new\vendor\php-http\client-common\src\HttpClientDecorator.php(30): Buzz\Client\FileGetContents->sendRequest(Object(Nyholm\Psr7\Request))
#1 G:\Server\xampp\htdocs\dev\mgapi.new\vendor\php-http\client-common\src\PluginClient.php(84): Http\Client\Common\EmulatedHttpAsyncClient->sendRequest(Object(Nyholm\Psr7\Request))
#2 G:\Server\xampp\htdocs\dev\mgapi.new\vendor\php-http\client-common\src\Plugin\HistoryPlugin.php(39): Http\Client\Common\PluginClient->Http\Client\Common\{closure}(Object(Nyholm\Psr7\Request))
#3 G:\Server\xampp\htdocs\dev\mgapi.new\vendor\php-http\client-common\src\PluginChain.php(48): Http\Client\Common\Plugin\HistoryPlugin->handleRequest(Object(Nyholm\Psr7\Req in G:\Server\xampp\htdocs\dev\mgapi.new\vendor\mailgun\mailgun-php\src\Exception\HttpServerException.php on line 28

Fatal error: Uncaught Buzz\Exception\NetworkException: file_get_contents(https://api.mailgun.net/v3/sandbox68cae6f5cd0443838a8851b2ceff97cb.mailgun.org/messages): failed to open stream: operation failed in G:\Server\xampp\htdocs\dev\mgapi.new\vendor\mailgun\mailgun-php\src\Exception\HttpServerException.php on line 28

Mailgun\Exception\HttpServerException: Mailgun's servers are currently unreachable. in G:\Server\xampp\htdocs\dev\mgapi.new\vendor\mailgun\mailgun-php\src\Exception\HttpServerException.php on line 28

Call Stack:
    0.1907     393696   1. {main}() G:\Server\xampp\htdocs\dev\mgapi.new\newlib.php:0
    0.3672    1195736   2. Mailgun\Api\Message->send() G:\Server\xampp\htdocs\dev\mgapi.new\newlib.php:22
    0.3736    1448856   3. Mailgun\Api\Message->httpPostRaw() G:\Server\xampp\htdocs\dev\mgapi.new\vendor\mailgun\mailgun-php\src\Api\Message.php:60
```

Kind REgards

Richard Handford

**Comment by DavidGarciaCat** [comment]
Hello, Richard,

Given the error message says:

```
Mailgun\Exception\HttpServerException: Mailgun's servers are currently unreachable
```

Then I should say that checking your internet connection and any potential firewall blocking connections is the first step.

I am not on my machine now, so it's a bit hard for me to check the code, but if you can confirm there aren't firewall rules affecting, then I would suggest checking with another HTTP client, just to discard an issue with Buzz.

You should be able to pick up any of the clients listed here: http://docs.php-http.org/en/latest/clients.html

Please keep me posted.

D

**Comment by rhandford** [comment]
David,

 

Thank you for your prompt reply.  

 

I have been using the api since May to monitor and report on Mailgun which has been great with 3.0 api.  Today I ran composer update and I am get this error.

 

I have disable the firewall but that has not resolved it.  I will check with another http client over the next couple of days and get back to you.

 

Thank you for your help it is much appreciated.

 

Kind Regards

 

Richard Handford

**Comment by rhandford** [comment]
David,

I have built the librairies using composer for the curl client.  This worked with my code.

So  I rebuilt the libraries for the buzz-adapter in a new directory and it runs fine with my code.

So I can only apologise for raising this issue, I am not sure what the issue was, but by rebuilding the libraries for the buzz adaptor seems to resolve the problem, so I have closed this issue.

Again thankyou for your help.

Kind regards

Richard Handford
